src/briefcase/commands/base.py

```diff
@@ -876,16 +871,39 @@ def update_cookiecutter_cache(self, template: str, branch="master"):
         if is_repo_url(template):
             # The app template is a repository URL.
             #
-            # When in `no_input=True` mode, cookiecutter deletes and reclones
-            # a template directory, rather than updating the existing repo.
+            # When in `no_input=True` mode, cookiecutter deletes and reclones a template
+            # directory, rather than updating the existing repo.
             #
-            # Look for a cookiecutter cache of the template; if one exists,
-            # try to update it using git. If no cache exists, or if the cache
-            # directory isn't a git directory, or git fails for some reason,
-            # fall back to using the specified template directly.
-            cached_template = cookiecutter_cache_path(template)
+            # Look for a Briefcase cache of the template.
+            cached_template = self.template_cache_path(template)
+
+            if cached_template.exists():
+                # There is a pre-existing cache of the template. Attempt to update it;
+                # if the fetch fails, use the existing state of the cache. Any other
+                # failure is surfaced to the user.
+                try:
+                    repo = self.tools.git.Repo(cached_template)
+                except self.tools.git.exc.GitError as e:
+                    raise BriefcaseCommandError(
+                        f"Template repository is in a weird state. Delete {cached_template} and retry."
+                    ) from e
+            else:
+                # This is the first time seeing this template. Perform a shallow clone.
+                try:
+                    self.logger.info(f"Cloning template {template!r}...")
+                    cached_template.mkdir(exist_ok=True, parents=True)
+                    repo = self.tools.git.Repo.clone_from(
+                        url=template,
+                        to_path=cached_template,
+                        filter=["blob:none"],
+                    )
```

Comment:
Of note, I think the fallback to "full clone" happens because I think that Git just sends the `filter=blob:none` to the Git server to see if it will support and respect filter....if the server rejects the filter for whatever reason, then I think the Git client just falls back the default behavior.

Reply:
If the fallback behavior is "full clone", and the version required is 6 years old, I'm comfortable opting into this behavior as the default.